### PR TITLE
API Update

### PR DESCRIPTION
**Pull request by fjebaker**
Updated scripts and code for the new caching API.

### CODE REVIEW

File: scripts/histogram_test.jl [hunk 1]
```diff
@@ -8,106 +8,113 @@ using Printf
...
+        u,
+        2000.0,
+        d,
+        fov_factor = fov * size_multiplier,
+        abstol = tolerance,
+        reltol = tolerance,
+        image_width = 350 * size_multiplier,
+        image_height = 350 * size_multiplier,
+        dtmax = dtmax,
+    )
```
Comment (fjebaker) [inline scripts/histogram_test.jl:63]: Using the caching API here!

File: scripts/histogram_test.jl [hunk 2]
```diff
@@ -8,106 +8,113 @@ using Printf
...
     )
 
     redshift_vf = (
-        ValueFunction(AccretionFormulae.redshift)
-        ∘ FilterValueFunction((m, sol, max_time; kwargs...) -> sol.u[end][2] > R_isco, NaN)
-        ∘ ConstValueFunctions.filter_early_term   
+        PointFunction(AccretionFormulae.redshift) ∘
+        FilterPointFunction((m, gp, max_time; kwargs...) -> gp.u[2] > R_isco, NaN) ∘
+        ConstPointFunctions.filter_early_term
     )
```
Comment (fjebaker) [inline scripts/histogram_test.jl:82]: Applying the point functions here!

**Comment by fjebaker** [comment]
@phajy @kyle8mcknight @dangoh123 see here.

The diffs are a little difficult to see, since I also ran the Julia style formatter over the repository to make things a bit more standard to read.

I've left review comments on the main changes -- please update your remotes when convenient.

**Comment by fjebaker** [comment]
The plot produced by the code I pushed looks like this:
plot_95

Just to compare it with Andy's script:
plot_107

where the `y` axis has been arbitrarily rescaled. Also I got rid of the check to see if we're within a certain radius, since the filter function now only returns a real number if the geodesic terminated through disc intersection, so shouldn't need to do that check anymore.

**Comment by phajy** [comment]
Looks good! Thanks for sorting out the cache and garbage collection.

Would it be helpful if I generated some test line profiles from XSPEC that we can use for testing purposes?

**Comment by fjebaker** [comment]
@phajy that would be brilliant!

**Pull request merged**